assets/js/app.js

```diff
@@ -0,0 +1,135 @@
+function collectData() {
+    const description = document.getElementById('description').value
+    const date = document.getElementById('date').value
+    const time = document.getElementById('time').value
+    const index = getNumberOfTasksInLocalStorage()
+    return {
+        index,
+        description,
+        date,
+        time,
+    }
+}
+
+function generateHTML(data) {
+    const newHTML = `
+        <div class="task" id="task-${data.index}" >
+        <i class="bi bi-x-circle" onclick="deleteTask(${data.index})"></i>
```

Comment:
if this `<i>` is wrapped inside `<div>&nbsp;<i>...</i></div>` then the flickering that is caused by the hovering is solved.